### PR TITLE
Implement Statistics Class

### LINKED ISSUE #125: Implementing the statistics class

### LINKED ISSUE #140: Calculate occupation rate of the network

### PR DESCRIPTION
**Pull request by X-Brast**
closes #125, closes #140

### CODE REVIEW

File: src/main/java/ch/heigvd/sitr/statistics/Statistics.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
+package ch.heigvd.sitr.statistics;
```
Comment (Laykel) [inline src/main/java/ch/heigvd/sitr/statistics/Statistics.java]: We usually write a small header in our file. See [the guidelines](https://github.com/HEIG-Boom/heigvd-pro-a-02-SiTR/wiki/Style-Guidelines)

File: src/main/java/ch/heigvd/sitr/statistics/Statistics.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+import ch.heigvd.sitr.gui.simulation.SimulationWindow;
+import ch.heigvd.sitr.vehicle.Vehicle;
+import java.util.LinkedList;
+
+/**
+ * Allows you to store simulation information in order to make statistics
+ */
```
Comment (Laykel) [inline src/main/java/ch/heigvd/sitr/statistics/Statistics.java]: Add `@author Alexandre Marques` to comply with guidelines.

File: src/main/java/ch/heigvd/sitr/statistics/Statistics.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+     * temporary
+     *
+     * @param occupancy The percentage
+     */
+    public void addNetworkOccupancy(int occupancy) {
+        if (occupancy >= 0 && occupancy <= 100)
+            networkOccupancy.add(occupancy);
+    }
+
+    /**
```
Comment (Laykel) [inline src/main/java/ch/heigvd/sitr/statistics/Statistics.java]: Would be great to add a small description.

File: src/main/java/ch/heigvd/sitr/statistics/Statistics.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            return 0;
+
+        double average = 0;
+        for(Vehicle v : vehicles){
+            average += v.getWaitingTime();
+        }
+        return average / vehicles.size();
+    }
+
+    /**
```
Comment (Laykel) [inline src/main/java/ch/heigvd/sitr/statistics/Statistics.java]: Would be great to add a small description.

File: src/main/java/ch/heigvd/sitr/statistics/Statistics.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+     */
+    public int getAccident(){
+        int nbAccident = 0;
+        for(Vehicle v : vehicles){
+            nbAccident += v.getNbAccidents();
+        }
+        return nbAccident;
+    }
+
+    /**
```
Comment (Laykel) [inline src/main/java/ch/heigvd/sitr/statistics/Statistics.java]: Would be great to add a small description.

File: src/main/java/ch/heigvd/sitr/statistics/Statistics.java (hunk 1 above)
Comment (X-Brast) [reply src/main/java/ch/heigvd/sitr/statistics/Statistics.java]: I did it

File: src/main/java/ch/heigvd/sitr/model/Simulation.java [hunk 6]
```diff
@@ -187,6 +197,8 @@ public void parseOpenDriveXml(RoadNetwork roadNetwork, String openDriveFilename)
      */
     public void stopLoop() {
         timer.cancel();
+        // Stop the Thread of statistics
+        stats.terminate();
```
Comment (Laykel) [inline src/main/java/ch/heigvd/sitr/model/Simulation.java]: I don't think we should start and stop the statistics like that. 

The problem here is that we need to be able to stop the loop and then restart it as many times as we want...

**Comment by Laykel** [comment]
This should also close #140, by the way.

**Pull request closed without merging**